ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/sendtransaction/TransactionTransmitter.java

```diff
@@ -74,7 +69,7 @@ private void createSignedTransactionBody() {
     final String signedTransactionHexString;
     try {
       if (!transaction.isNonceUserSpecified()) {
```

Comment:
is this conditional going to move into the Transaction?

Reply:
Unfortunately can't easily do that. Most obvious place would be the rlpEncode function in transaction but this is called multiple times by the serialiser. Other possible way would to move the serialiser into the transaction and use to fully create the json request complete with the encoded string but think this going a bit far for what I think the transaction is at the moment.
So in summary, needs to stay where it is until we come with a better idea.